gcloud-java-examples/src/main/java/com/google/gcloud/examples/dns/DnsExample.java

```diff
@@ -0,0 +1,526 @@
+        while (zoneIterator.hasNext()) {
+          Zone zone = zoneIterator.next();
+          printZone(zone);
+        }
+      } else {
+        System.out.println("Project contains no zones.");
+      }
+    }
+
+    @Override
+    public String params() {
+      return "";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 0;
+    }
+  }
+
+  private static class GetZoneAction implements DnsAction {
+
+    /**
+     * Gets details about a zone with the given name.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      Zone zone = dns.getZone(zoneName);
+      if (zone == null) {
+        System.out.printf("No zone with name '%s' exists.%n", zoneName);
+      } else {
+        printZone(zone);
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 1;
+    }
+  }
+
+  private static class DeleteZoneAction implements DnsAction {
+
+    /**
+     * Deletes a zone with the given name.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      boolean deleted = dns.delete(zoneName);
+      if (deleted) {
+        System.out.printf("Zone %s was deleted.%n", zoneName);
+      } else {
+        System.out.printf("Zone %s was NOT deleted. It does not exist.%n", zoneName);
+      }
+    }
+
+    @Override
+    public String params() {
+      return "<zone_name>";
+    }
+
+    @Override
+    public boolean check(String... args) {
+      return args.length == 1;
+    }
+
+  }
+
+  private static class DeleteDnsRecordAction implements DnsAction {
+
+    /**
+     * Deletes a DNS record of type A from the given zone. The last parameter is ttl and it is not
+     * required. If ttl is not provided, a default value of 0 is used. The service requires a
+     * precise match (including ttl) for deleting a record.
+     */
+    @Override
+    public void run(Dns dns, String... args) {
+      String zoneName = args[0];
+      String recordName = args[1];
+      String ip = args[2];
+      int ttl = 0;
+      if (args.length > 3) {
+        ttl = Integer.parseInt(args[3]);
+      }
+      DnsRecord record = DnsRecord.builder(recordName, DnsRecord.Type.A)
+          .records(ImmutableList.of(ip))
+          .ttl(ttl, TimeUnit.SECONDS)
+          .build();
+      ChangeRequest changeRequest = ChangeRequest.builder()
+          .delete(record)
+          .build();
+      changeRequest = dns.applyChangeRequest(zoneName, changeRequest);
+      System.out.printf("The request for deleting A record %s for zone %s was successfully " +
+          "submitted and assigned ID %s.%n", recordName, zoneName, changeRequest.id());
+      System.out.print("Waiting for deletion to happen...");
+      while (changeRequest.status().equals(ChangeRequest.Status.PENDING)) {
```

Comment:
This while loop is also used below in `AddDnsRecordAction`.  Can we factor this code into a method?